nix/package.nix

```diff
@@ -0,0 +1,23 @@
+{ buildGoModule, lib }:
+
+buildGoModule {
+  name = "socketmaster";
+  src = lib.cleanSourceWith {
+    src = lib.cleanSource ../.;
+    filter = path: type:
+      baseNameOf path != "nix" && (
+        lib.hasSuffix ".go" path
+        || lib.hasSuffix "/go.mod" path
+        || lib.hasSuffix "/go.sum" path
+      );
+  };
+
+  vendorSha256 = "sha256-bPIzk+g8lxntEREAp1ZJiHTGxXFRybtZjhR23uGIFR4=";
```

Comment:
I remember being able to set this to null if all the code was vendored. It might be nicer to use rather than to update that hash all of the time.

Reply:
That will require the vendor directory to be committed.

Reply:
Fine by me

Reply:
Hmm, getting an error, and go's suggestion doesn't work.

<details>

```
  # attribute must be present
  vendorSha256 = `null`;
```

```
go: inconsistent vendoring in /build/source:
        google.golang.org/grpc@v1.45.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        gopkg.in/yaml.v3@v3.0.0-20220512140231-539c8e751b99: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        github.com/golang/protobuf@v1.5.2: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        golang.org/x/net@v0.0.0-20200822124328-c89045814202: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        golang.org/x/sys@v0.0.0-20200323222414-85ca7c5b95cd: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        golang.org/x/text@v0.3.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        google.golang.org/genproto@v0.0.0-20200526211855-cb27e3aa2013: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
        google.golang.org/protobuf@v1.26.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt

        To ignore the vendor directory, use -mod=readonly or -mod=mod.
        To sync the vendor directory, run:
                go mod vendor
```

</details>

> update that hash all of the time.

When dependencies change, but not when the local code changes, fwiw.

Reply:
Ok, this is secondary so feel free to ignore